src/HotChocolate/Core/src/Types/Types/Scalars/FloatTypeBase.cs

```diff
@@ -96,7 +96,7 @@ protected virtual bool IsInstanceOfType(TRuntimeType value)
                 return null;
             }
 
-            if (valueSyntax is FloatValueNode floatLiteral)
+            if (valueSyntax is FloatValueNode floatLiteral && IsInstanceOfType(floatLiteral))
```

Comment:
@michaelstaib  this is a change in the base scalar, is this correct? 
Added also the error factories and possibliy fixed a bug